vm/src/stdlib/os.rs

```diff
@@ -1542,6 +1542,50 @@ mod posix {
             .map_err(|err| err.into_pyexception(vm))
     }
 
+    #[pyfunction]
+    fn execve(
+        path: PyPathLike,
+        args: Either<PyListRef, PyTupleRef>,
+        env: PyDictRef,
+        vm: &VirtualMachine,
+    ) -> PyResult<()> {
+        let path = ffi::CString::new(path.into_bytes())
+            .map_err(|_| vm.new_value_error("embedded null character".to_owned()))?;
+
+        let args: Vec<ffi::CString> = match args {
+            Either::A(list) => vm.extract_elements(list.as_object())?,
+            Either::B(tuple) => vm.extract_elements(tuple.as_object())?,
+        };
+        let args: Vec<&ffi::CStr> = args.iter().map(|entry| entry.as_c_str()).collect();
+
+        let first = args
+            .first()
+            .ok_or_else(|| vm.new_value_error("execv() arg 2 must not be empty".to_owned()))?;
+
+        if first.to_bytes().is_empty() {
+            return Err(vm.new_value_error("execv() arg2 first element cannot be empty".to_owned()));
+        }
+
+        let mut env_vec = vec![];
+        for (k, v) in env {
+            let key: PyStringRef = k.downcast().map_err(|_| {
```

Comment:
`TryFromObject` is fitting here. Probably

```rust
let key = PyPathLike::try_from_object(k)?
```

Reply:
Thanks for the info. I'd gladly try that and make a commit

Reply:
fix is up. https://github.com/RustPython/RustPython/pull/2101/commits/b6621913921af0e7adfa762b5fa8a30cec9911df